### PR TITLE
Added vector search for create collection

### LINKED ISSUE #455: Create collection enhancement to support vector search configuration
Vector search index for a collection to be enabled while creating a collection. A vector clause value can be provided in the options. The create collection operation will create the physical table in Cassandra with a query_vector_value column of type VECTOR<FLOAT, n> where n is the size of the vector when n > 0. A vector search index is created for the query_vector_value column. Only one vector field per collection will be allowed.

```
API: /v1/{namespace}
{
  "createCollection" : {
    "name" : "vector_collection"
    "options" : {
      "vectors" : {
	  "size" : 4,
         "function" : "cosine" # can be one of the following [dot_product|euclidean|cosine(default)]
      }
    }
  }
}

```

CQL for vector index creation:
`
CREATE CUSTOM INDEX IF NOT EXISTS ann_index ON collection(query_vector_value) USING 'StorageAttachedIndex' WITH OPTIONS = { 'similarity_function': 'cosine'}; `

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Added vector search in create collection command.
**Which issue(s) this PR fixes**:
Fixes #455 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -107,6 +107,7 @@ jobs:
       - name: Integration Test
         env:
           COORD_IMAGE: 237073351946.dkr.ecr.us-east-1.amazonaws.com/stargateio/coordinator-dse-68
+          TEST_REGEX: '!VectorSearchIntegrationTest*'
```
Comment (tatu-at-datastax) [inline .github/workflows/continuous-integration.yaml:110]: This is not being used, I think need to add to below:

     -Dit.test=$TEST_REGEX 

after `-DskipUnitTests` ?

Comment (maheshrajamani) [reply .github/workflows/continuous-integration.yaml:110]: For testing I removed it temporarily

Comment (tatu-at-datastax) [reply .github/workflows/continuous-integration.yaml:110]: Ah I should have looked at commits. It wouldn't have skipped the test etc.

**Review by vkarpov15 — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java [hunk 2]
```diff
@@ -15,5 +14,39 @@ public record CreateCollectionCommand(
...
+              type = SchemaType.OBJECT,
+              implementation = VectorSearchConfig.class)
+          VectorSearchConfig vector) {
+
+    public record VectorSearchConfig(
+        @Schema(
+                description = "Vector field embedding size",
+                type = SchemaType.INTEGER,
+                implementation = Integer.class)
+            Integer size,
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:39]: Should we add annotations to ensure this as required, and with minimum (1) and maximum (? do we know this) values?

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java [hunk 3]
```diff
@@ -15,5 +14,39 @@ public record CreateCollectionCommand(
...
+        @Nullable
+            @Pattern(
+                regexp = "(dot_product|cosine|euclidean)",
+                message = "function name can only be 'dot_product', 'cosine' or 'euclidean'")
+            @Schema(
+                description =
+                    "Similarity function algorithm that needs to be used for vector search",
+                defaultValue = "cosine",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:47]: Is "cosine" default that CQL has? (I think we want to use same as CQL default, if there is one)

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java [hunk 4]
```diff
@@ -63,7 +63,9 @@ public enum ErrorCode {
 
   UNSUPPORTED_UPDATE_OPERATION_TARGET("Unsupported target JSON value for update operation"),
 
-  UNSUPPORTED_UPDATE_FOR_DOC_ID("Cannot use operator with '_id' field");
+  UNSUPPORTED_UPDATE_FOR_DOC_ID("Cannot use operator with '_id' field"),
+
+  VECTOR_SEARCH_NOT_AVAILABLE("Vector search is not enabled in the backend");
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:68]: Vector search -> Vector search functionality
enabled -> available?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java [hunk 5]
```diff
@@ -34,6 +49,23 @@ public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
 
   protected QueryOuterClass.Query getCreateTable(String keyspace, String table) {
     String createTable =
+        "CREATE TABLE IF NOT EXISTS \"%s\".\"%s\" ("
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java:52]: Move these inside respective "if" / "else" branches?

**Review by tatu-at-datastax — APPROVED** [review]
LGTM: minor suggestions to consider.

**Pull request merged**